src/framework/mlt_frame.c

```diff
@@ -445,6 +470,9 @@ int mlt_image_format_size( mlt_image_format format, int width, int height, int *
 		case mlt_image_glsl_texture:
 			if ( bpp ) *bpp = 0;
 			return 4;
+		case mlt_image_yuv422p16:
+			if ( bpp ) *bpp = 2;
+			return height * ( UP16( width * 2 ) + 2 * UP16( width + 1 ) ) ;
```

Comment:
I don't think this function is intended to return SMID optimized sizes. It doesn't do that for any of the other formats. @ddennedy do you have a preference about whether this function can return a number that could possibly be greater than the size actually needed to store the image?